dotCMS/src/main/java/com/dotmarketing/business/VersionableFactoryImpl.java

```diff
@@ -476,7 +452,23 @@ private List<ContentletVersionInfo> findContentletVersionInfos(final String iden
 	@Override
 	protected List<ContentletVersionInfo> findAllContentletVersionInfos(final String identifier)
 			throws DotDataException, DotStateException {
-		return findContentletVersionInfos(identifier, -1);
+		if(identifier==null){
+			throw new DotRuntimeException("identifier cannot be null");
+		}
+
+		List<ContentletVersionInfo> infos = icache.getContentVersionInfos(identifier);
+		if (infos != null) {
+			return infos;
+		}
+		synchronized (identifier.intern()) {
```

Comment:
I think we need to address 

https://sonarqube.dotcms.site/project/issues?pullRequest=26968&resolved=false&severities=BLOCKER%2CCRITICAL%2CMAJOR%2CMINOR&types=BUG&open=AYxawmUXwfF26KX8_jpm&id=dotCMS_core_AYSbIemxK43eThAXTlt-

We should use the a Striped lock mechanism, something like this:

https://www.baeldung.com/java-lock-stripping#bd-4-concurrent-access-with-striped